src/modules/flow/hc-sr04/hc-sr04.json

```diff
@@ -0,0 +1,78 @@
+{
+  "$schema": "http://solettaproject.github.io/soletta/schemas/node-type-genspec.schema",
+  "name": "hc-sr04",
+  "meta": {
+    "author": "Intel Corporation",
+    "license": "Apache-2.0",
+    "version": "1"
+  },
+  "types": [
+    {
+      "category": "input/hw",
+      "description": "Ultrasonic ranging module HC-SR04. Provides non-contact distance measurements, including ultrasonic transmitters, receiver and control circuit.",
+      "in_ports": [
+        {
+          "data_type": "any",
+          "description": "Send a ultrasonic wave and measure distance to obstacle. After each pulse a distance will be calculated and sent on output port.",
+          "methods": {
+            "process": "trigger_process"
+          },
+          "name": "TRIGGER"
+        }
+      ],
+      "methods": {
+        "close": "hc_sr04_close",
+        "open": "hc_sr04_open"
+      },
+      "name": "hc-sr04/distance",
+      "options": {
+        "members": [
+          {
+            "data_type": "string",
+            "description": "Label of the 'Trigger' pin on the board. This pin will receive a pulse to emit the wave. If raw is set to true, this should be the pin number as recognized by the platform.",
+            "name": "trigger"
+          },
+          {
+            "data_type": "boolean",
+            "default": false,
+            "description": "Change 'trigger' meaning to be the system parameters needed to address the desired pin. Use it if you know what you are doing.",
+            "name": "trigger_raw"
+          },
+          {
+            "data_type": "string",
+            "description": "Label of the 'Echo' pin on the board. This pin will read a pulse and based on its width distance is calculated. If raw is set to true, this should be the pin number as recognized by the platform.",
+            "name": "echo"
+          },
+          {
+            "data_type": "boolean",
+            "default": false,
+            "description": "Change 'echo' meaning to be the system parameters needed to address the desired pin. Use it if you know what you are doing.",
+            "name": "echo_raw"
+          },
+          {
+            "data_type": "int",
+            "default": 30,
+            "description": "Polling time on 'echo' pin.",
+            "name": "echo_poll_timeout"
+          },
+          {
+            "data_type": "int",
+            "default": 400,
+            "description": "Offset between pulse on trigger end and pulse on echo starts, on microseconds. It's only required if the echo pulse is too short that when the first interruption is handled pulse ended already.",
```

Comment:
"Offset between pulse on trigger end and pulse on echo starts, on microseconds. It's only required if the echo pulse is too short that when the first interruption is handled pulse ended already." I'd rephrase as "An additional time offset between the pulse on trigger and the echo response events, on microseconds. It's only required if the echo response is so fast that turning off the trigger happens after that. It may happen on non real time systems, like Linux" or something like that.